src/components/LoadingIndicator/LoadingIndicator.test.tsx

```diff
@@ -1,48 +1,112 @@
+enum MockSpinnerPositionType {
+  BOTTOM = 'bottom',
+  LEFT = 'left',
+  RIGHT = 'right'
+}
 
-const mockText = 'LoadingIndicator Text';
-const mockCustomClassName = 'my_custom--class-name';
+const MOCK_DEFAULT_PROPS = {
+  className: 'mock-class-name',
+  text: 'Mock Text',
+  spinnerPosition: MockSpinnerPositionType.LEFT
+};
 
-test('should render container', () => {
-  const { getByTestId } = renderComponent({});
-  const container = getByTestId(TEST_ID.CONTAINER);
-  expect(container).toBeInTheDocument();
-  expect(container).toHaveClass('zui-loading-indicator');
-  expect(container).toHaveClass('zui-loading-indicator-bottom');
-});
+const MOCK_SPINNER = 'Mock Spinner';
 
-test('should render container with custom class name', () => {
-  const { getByTestId } = renderComponent({ className: mockCustomClassName });
-  const container = getByTestId(TEST_ID.CONTAINER);
-  expect(container).toHaveClass(mockCustomClassName);
-});
+jest.mock('./Spinner', () => ({
+  Spinner: () => <div>{MOCK_SPINNER}</div>
+}));
 
-test('should render container with custom position', () => {
-  const { getByTestId } = renderComponent({ spinnerPosition: 'right' });
-  const container = getByTestId(TEST_ID.CONTAINER);
-  expect(container).toHaveClass('zui-loading-indicator-right');
-});
+const renderComponent = (props?: LoadingIndicatorProps) =>
+  render(
+    <LoadingIndicator
+      className={MOCK_DEFAULT_PROPS.className}
+      text={MOCK_DEFAULT_PROPS.text}
+      spinnerPosition={MOCK_DEFAULT_PROPS.spinnerPosition}
+      {...props}
+    />
+  );
 
-test('should render spinner', () => {
-  const { getByTestId } = renderComponent({});
-  const spinner = getByTestId(TEST_ID.SPINNER);
-  expect(spinner).toBeInTheDocument();
-});
+describe('LoadingIndicator', () => {
+  test('should render loading indicator container', () => {
+    const { container } = renderComponent();
+    const loadingIndicatorContainer = container.getElementsByClassName('zui-loading-indicator');
 
-test('should render text', () => {
-  const { getByTestId } = renderComponent({ text: mockText });
-  const text = getByTestId(TEST_ID.TEXT);
-  expect(text).toBeInTheDocument();
-  expect(text).toHaveClass('zui-loading-indicator-text');
-  expect(text).toHaveClass('zui-loading-indicator-text-bottom');
-});
+    expect(loadingIndicatorContainer.length).toBe(1);
+  });
+
+  test('should render loading indicator text container and text value', () => {
+    const { container } = renderComponent();
+    const textContainer = container.getElementsByClassName('zui-loading-indicator-text');
+
+    expect(textContainer.length).toBe(1);
+    expect(textContainer[0].textContent).toEqual(MOCK_DEFAULT_PROPS.text);
+  });
+
+  test('should render spinner component', () => {
+    const { getByText } = renderComponent();
+
+    expect(getByText(MOCK_SPINNER)).toBeTruthy();
+  });
+
+  describe('when props are defined', () => {
```

Comment:
These describe's could maybe be clearer if you like? Noticing others don't really use these, happy to remove if you want, I quite like them in the logs when running tests

Reply:
`describe` is a handy tool, because you can create scoped context which can reduce duplicated code, e.g.
```
describe('something', () => {
    beforeEach(() => {
        // do some setup here that all tests use
    })

    test('something related', () => {
        // use context created by local beforeEach
    })
})
```
`describe` also just creates better signposting - easier to find the test you're looking for if they're grouped nicely. We should probably be using `describe` more often.

I think the `describe`s in this test could be improved. Perhaps grouping them by what feature is being tested would be cleaner - `describe('spinner positioning', ...)`, `describe('class names', ...)`. If you have a look at [the Jest docs here](https://jestjs.io/docs/setup-teardown#scoping), they have `describe('matching cities to foods', ...)` which is a nice high level description of the feature they're testing.

Reply:
Sweet thanks will check this out and update